docs/getting_started.rst

```diff
@@ -1,135 +1,288 @@
+        {
+            "A": 5.0,
+            "B": pd.Timestamp("20110102"),
+            "C": pd.Series(1, index=list(range(4)), dtype="float32"),
+            "D": np.array([12] * 4, dtype="int32"),
+            "E": pd.Categorical(["prod", "train", "test", "train"]),
+            "F": "bar",
+        }
+    )
+    another_df
+    df.dtypes
+    another_df.dtypes  # both have the same schema
 
-After we have written the data, we want to read it back in again. For this we
-use :func:`kartothek.io.eager.read_table`. This method
-returns the whole dataset as a pandas DataFrame and the metadata of the
-dataset. The metadata of a dataset is a dict where one can store arbitrary
-information about the dataset. As this metadata is always loaded on accessing
-the dataset, this should be kept small.
+    store_dataframes_as_dataset(
+        store_factory, "another_unique_dataset_identifier", [df, another_df]
+    )
 
 
+However, passing a list of dataframes with differing schemas `without specifying table names`
+to :func:`~kartothek.io.eager.store_dataframes_as_dataset` throws ``ValueError``:
+
 .. ipython:: python
-   :okwarning:
 
-   from kartothek.io.eager import read_table
+    df2 = pd.DataFrame(
+        {
+            "G": "foo",
+            "H": pd.Categorical(["test", "train", "test", "train"]),
+            "I": np.array([9] * 4, dtype="int32"),
+            "J": pd.Series(3, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190604"),
+            "L": 2.0,
+        }
+    )
+    df2
+    df.dtypes
+    df2.dtypes  # schema is different!
+
+
+.. ipython::
+
+    @verbatim
+    In [24]: store_dataframes_as_dataset(
+       ....:     store_factory,
+       ....:     "will_not_work",
+       ....:     [df, df2],
+       ....: )
+       ....:
+    ---------------------------------------------------------------------------
+    ValueError: Schema violation
+    Origin schema: {table/9e7d9217c82b4fda9c4e720dc987c60d}
+    Origin reference: {table/80feb4d84ac34a9c9d08ba48c8170647}
+
+
+.. note:: Read these sections for more details: :ref:`type_system`, :ref:`dataset_spec`
+
+
+When we do not explicitly define the name of the table and partition, ``kartothek`` uses the
+default table name ``table`` and generates a UUID for the partition name.
+
+.. admonition:: A more complex example: multiple named tables
+
+    Sometimes it may be useful to write multiple dataframes with different schemas into
+    a single dataset. This can be achieved by creating a dataset with multiple tables.
+
+    In this example, we create a dataset with two tables: ``core-table`` and ``aux-table``.
+    The schemas of the tables are identical across partitions (each dictionary in the
+    ``dfs`` list argument represents a partition).
+
+    .. ipython:: python
+
+        dfs = [
+            {
+                "data": {
+                    "core-table": pd.DataFrame({"col1": ["x"]}),
+                    "aux-table": pd.DataFrame({"f": [1.1]}),
```

Comment:
I recommend to use a common column for the two tables. Storing data this way hardly offers any benefits over storing the data in two separate datasets. OTOH, if you have a common column this is an opportunity for trivial join logic. I wouldn't necessarily mention this in a getting started guide, though.